model/AbstractRdsResultStorage.php

```diff
@@ -0,0 +1,727 @@
+    {
+        $variableData = $this->prepareVariableData($deliveryResultIdentifier, $test, $variable, $callId);
+        $variableData[self::CALL_ID_TEST_COLUMN] = $callId;
+        return $variableData;
+    }
+
+    /**
+     * Prepares data to be inserted in database.
+     *
+     * @param string   $deliveryResultIdentifier
+     * @param string   $test
+     * @param Variable $variable
+     * @param string   $callId
+     *
+     * @return array
+     */
+    protected function prepareVariableData($deliveryResultIdentifier, $test, Variable $variable, $callId)
+    {
+        // Ensures that variable has epoch.
+        if (!$variable->isSetEpoch()) {
+            $variable->setEpoch(microtime());
+        }
+
+        return $this->prepareVariableDataForSchema($deliveryResultIdentifier, $test, $variable, $callId);
+    }
+
+    /**
+     * Prepares data to be inserted in database according to a given schema.
+     *
+     * @param string   $deliveryResultIdentifier
+     * @param string   $test
+     * @param Variable $variable
+     * @param string   $callId
+     *
+     * @return array
+     */
+    abstract protected function prepareVariableDataForSchema($deliveryResultIdentifier, $test, Variable $variable, $callId);
+
+    /**
+     * Builds a variable from database row.
+     *
+     * @param array $variable
+     *
+     * @return \stdClass
+     */
+    protected function getResultRow($variable)
+    {
+        $resultVariable = $this->unserializeVariableValue($variable[self::VARIABLE_VALUE]);
+        $object = new \stdClass();
+        $object->uri = $variable[self::VARIABLES_TABLE_ID];
+        $object->class = get_class($resultVariable);
+        $object->deliveryResultIdentifier = $variable[self::VARIABLES_FK_COLUMN];
+        $object->callIdItem = $variable[self::CALL_ID_ITEM_COLUMN];
+        $object->callIdTest = $variable[self::CALL_ID_TEST_COLUMN];
+        $object->test = $variable[self::TEST_COLUMN];
+        $object->item = $variable[self::ITEM_COLUMN];
+        $object->variable = clone $resultVariable;
+
+        return $object;
+    }
+
+    /**
+     * @return QueryBuilder
+     */
+    protected function getQueryBuilder()
+    {
+        return $this->getPersistence()->getPlatform()->getQueryBuilder();
+    }
+
+    /**
+     * @return \common_persistence_SqlPersistence
+     */
+    public function getPersistence()
+    {
+        if ($this->persistence === null) {
+            $persistenceId = $this->hasOption(self::OPTION_PERSISTENCE) ?
+                $this->getOption(self::OPTION_PERSISTENCE)
+                : 'default';
+            $this->persistence = $this->getServiceLocator()->get(PersistenceManager::SERVICE_ID)->getPersistenceById($persistenceId);
+        }
+
+        return $this->persistence;
+    }
+
+    /**
+     * @param $value
+     *
+     * @return mixed
+     */
+    protected function unserializeVariableValue($value)
+    {
+        return unserialize($value);
```

Comment:
this is dangerous!

Reply:
That's the current implementation for which we have existing data. We cannot just change it in a snap. A discussion has happened with Joel and Camille on the matter, concluding to postpone the change to a json serialisation because of complexity and migration script needed.